### PR TITLE
Bugsnag User Identification

### PR DESCRIPTION
**Pull request by OSPFNeighbour**
React component BugSnagUserHandler to register user name against bugsnag

Test for new component

Added Alert to burger page to wrap around test bug report button to give the user some feedback

### CODE REVIEW

**Comment by OSPFNeighbour** [comment]
possible edge case where a user logs out and back in as a different user without closing the app and it will report as the first user... not sure how much I care

File: client/src/components/bugsnag/__tests__/BugSnagUserHandler.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import TestRenderer from 'react-test-renderer';
+import BugSnagUserHandler from '../BugSnagUserHandler';
+
+test('lifecycle', async () => {
+  const bugsnag = {
```
Comment (sdunster) [inline client/src/components/bugsnag/__tests__/BugSnagUserHandler.js]: you could use jest mocks - might make it a bit simpler
```
const bugsnag = {
  setUser: jest.fn(),
}
```
I'd also suggest putting this outside the test because it can then be defined in one place and used by all tests. All tests run with independent module globals so you don't need to worry about conflicts.

File: client/src/components/bugsnag/__tests__/BugSnagUserHandler.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    <BugSnagUserHandler
+      bugsnag={bugsnag}
+      auth={{ id: 1, username: 'test' }}
+    />,
+  );
+
+  await renderer.root.updatePromise;
+
+
+  expect(bugsnag.username).toEqual('test');
```
Comment (sdunster) [inline client/src/components/bugsnag/__tests__/BugSnagUserHandler.js]:
```
expect(bugsnag.setUser.mock.calls.length).toBe(0);
expect(bugsnag.setUser.mock.calls[0][0]).toEqual('1');
expect(bugsnag.setUser.mock.calls[0][1]).toEqual('test');
expect(bugsnag.setUser.mock.calls[0][2]).toEqual(undefined);
```

File: client/src/components/bugsnag/BugSnagUserHandler.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { isLoggedIn } from '../../selectors/auth';
+
+class BugSnagUserHandler extends React.Component {
+  constructor() {
+    super();
+    this.didRegister = false;
```
[the remark this answers is not in the document]
Comment (OSPFNeighbour) [reply client/src/components/bugsnag/BugSnagUserHandler.js]: didn't think you could have a class level varible. I get a syntax errors

File: client/src/components/bugsnag/__tests__/BugSnagUserHandler.js (hunk 1 above)
Comment (OSPFNeighbour) [reply client/src/components/bugsnag/__tests__/BugSnagUserHandler.js]: I put it outside the test and it seemed to not be independent, calls length didnt reset by the end of it

File: client/src/components/bugsnag/BugSnagUserHandler.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (sdunster) [reply client/src/components/bugsnag/BugSnagUserHandler.js]: no it definitely works: https://github.com/NSWSESMembers/availability-poc/blob/master/client/src/screens/auth/SignUp.js#L17

[the remark this answers is not in the document]
Comment (OSPFNeighbour) [reply client/src/components/bugsnag/BugSnagUserHandler.js]: oh you mean with state.... not a class variable.
